proposals/0132-sequence-end-ops.md

```diff
@@ -16,724 +17,227 @@ these names so they follow consistent, predictable patterns.
+| ---------------- | --------------------------------- | ----------------------------- |
+| Get value        | First element                     | `first`                       |
+| "                | Last element                      | `last`                        |
+| "                | First *n* elements                | `prefix(3)`                   |
+| "                | Last *n* elements                 | `suffix(3)`                   |
+| "                | Leading elements matching closure | `prefix(while: isOdd)`        |
+| "                | Earliest element matching closure | `first(where: isOdd)`         |
+| Get index        | Earliest element equal to value   | `index(of: x)`                |
+| "                | Earliest element matching closure | `index(where: isOdd)`         |
+| Remove from copy | First element                     | `dropFirst()`                 |
+| "                | Last element                      | `dropLast()`                  |
+| "                | First *n* elements                | `dropFirst(3)`                |
+| "                | Last *n* elements                 | `dropLast(3)`                 |
+| "                | Leading elements matching closure | `drop(while:)`                |
+| Remove from self | First element                     | `removeFirst()`               |
+| "                | Last element                      | `removeLast()`                |
+| "                | First *n* elements                | `removeFirst(3)`              |
+| "                | Last *n* elements                 | `removeLast(3)`               |
+| …if present      | First element                     | `popFirst()`                  |
+| "                | Last element                      | `popLast()`                   |
+| Test equality    | First *n* elements                | `starts(with: other)`         |
+| …with function   | "                                 | `starts(with: other, by: ==)` |
+
+Put next to each other, we see a lot of inconsistent terminology:
+
+* Usually, "first *n* elements" is handled by overloading a `first` method, except 
+  on the most-used category, "get value of element(s)". There, we suddenly 
+  use `prefix` and `suffix`.
+
+* The "get index" methods do not indicate a direction, but adding 
+  versions which search from the end would be very plausible. Similarly, 
+  `drop(while:)` does not include a direction, but dropping trailing elements is a plausible feature.
+
+* "Remove from copy" and "Remove from self" are 
+  closely related, but they have unrelated names. The name `drop`, while a 
+  term of art from functional languages, sounds like a mutating operation that 
+  deletes data; in particular, developers experienced with SQL may find "drop" 
+  alarming.
+
+* `starts(with:)` looks like nothing else in this list, even though it does 
+  similar things, and even though Foundation uses `hasPrefix(_:)`, which 
+  *does* sound like other entries in this table.
```

Comment:
It’s very possible this name could be improved, but I strongly disagree with almost all points in the bullet.

* It doesn’t do anything similar to other things in the list as far as I can tell
* Looking/sounding similar to other operations is not in and of itself a virtue, especially if its semantics are utterly different.  The idea that names should blend in with the environment of other names is one of the great fallacies of naming.

Using the word `prefix`, if it’s used elsewhere to mean the same thing, definitely has virtue.  However, given my arguments above about slice notation I’m inclined to think that we shouldn’t end up with any other uses of the term.